app/assets/javascripts/application.js

```diff
@@ -1,5 +1,60 @@
 //= require all.js
+//= require jquery
+//= require zxcvbn.js
 
 window.addEventListener("DOMContentLoaded", () => {
   window.GOVUKFrontend.initAll();
 });
+
+const getPasswordStrength = password => {
+  const result = zxcvbn(password);
+  const score = 100 * result.score / 4;
+  const scoreText = score < 50 ? 'weak' : (score < 100 ? 'medium' : 'good');
+  const rawSuggestion = result.feedback.suggestions.join(', ');
+  // zxcvbn-js sometimes fails to return a suggestion even when the password is too weak, so we need to perform some coercion.
+  const suggestion = (!rawSuggestion) ? 'Choose a password that is harder to guess' : rawSuggestion;
+  return { score, scoreText, suggestion };
+};
+
+if ($('input[name="user[password]"]').length) {
+  $('body').on('keyup', 'input[name="user[password]"]', function(e) {
+    const { score, scoreText } = getPasswordStrength(this.value)
+    const feedback = `Password strength: ${scoreText}`;
+    if (this.value.length) {
+      $('.password-strength')
+        .css('display', 'inline-block')
+        .css('background-position', score + '% 0%')
```

Comment:
Don't think this is needed anymore - it's probably for the fancy CSS transition thing I did earlier.